RealexPayments/HPP/Controller/Process/Result/Base.php

```diff
@@ -116,11 +127,34 @@ private function _handleResponse($response)
             return false;
         }
         //get the actual order id
-        list($incrementId, $orderTimestamp) = explode('_', $response['ORDER_ID']);
+        [$incrementId, $orderTimestamp] = explode('_', $response['ORDER_ID']);
 
         if ($incrementId) {
             $order = $this->_getOrder($incrementId);
             if ($order->getId()) {
+                // skip default response processing if APM
+                if ($this->_paymentManagement->isTransactionApm($response)) {
```

Comment:
I'd prefer reducing the nesting here by returning early instead of adding additional nested expressions. For example, something similar to the below:

```php
if (!$incrementId) {
    // log
    return false;
}

if (!$order->getId()) {
    // log
    return false;
}

if (!$this->_paymentManagement->isTransactionApm($response)) {
    // process the response
    return $this->_paymentManagement->processResponse($order, $response);
}

// handle apm case
```